zivid_camera/src/zivid_camera.cpp

```diff
@@ -403,7 +407,23 @@ bool ZividCamera::captureAssistantSuggestSettingsServiceHandler(CaptureAssistant
   capture_settings_controller_->setZividSettings(suggested_settings);
 
   return true;
-}  // namespace zivid_camera
+}
+
+bool ZividCamera::loadSettingsFromFileServiceHandler(LoadSettingsFromFile::Request& req,
+                                                     LoadSettingsFromFile::Response&)
+{
+  ROS_DEBUG_STREAM(__func__ << ": Request: " << req);
+  capture_settings_controller_->setZividSettings(Zivid::Settings{ req.file_path.c_str() });
+  return true;
```

Comment:
I guess since you are throwing inside `setZividSettings()` if setting the settings does not complete, you cannot answer a `false` bool on the load setting service.

Reply:
Exceptions are used for errors, also other places like for `capture`. The node does not shut down if a service function call throws, it is caught by something in ROS which makes the service call error out:

```
$ rosservice call /zivid_camera/load_settings_from_file "file_path: 'x'" 
ERROR: service [/zivid_camera/load_settings_from_file] responded with an error: b'Failed loading Settings from "x"'
$ rosservice call /zivid_camera/load_settings_from_file "file_path: 'y'" 
ERROR: service [/zivid_camera/load_settings_from_file] responded with an error: b'Failed loading Settings from "y"'
$ rosservice call /zivid_camera/load_settings_from_file "file_path: 'z'" 
ERROR: service [/zivid_camera/load_settings_from_file] responded with an error: b'Failed loading Settings from "z"'
```

```
[INFO] [1636467401.453709]: Calling capture service
[ INFO] [1636467401.457850638]: Capturing with 3 acquisition(s)
[INFO] [1636467403.330188]: PointCloud received
[INFO] [1636467403.344540]: 2D color image received
[ERROR] [1636468166.470776730]: Exception thrown while processing service call: Failed loading Settings from "x"
[ERROR] [1636468172.394322735]: Exception thrown while processing service call: Failed loading Settings from "y"
[ERROR] [1636468174.872632798]: Exception thrown while processing service call: Failed loading Settings from "z"
```

Reply:
Ok, that looks reasonable :+1: